File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        nums = sorted(nums)
        count = 1
        max_count = 1
        for i in range(len(nums) - 1):
            if nums[i+1] == nums[i]:
                continue
            elif nums[i+1] == nums[i] + 1:
                count += 1
                continue
            else:
                max_count = max(count, max_count)
                count = 1
        return max(count, max_count)

File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const sorted = nums.map((val, idx) => [val, idx]).sort((a, b) => a[0] - b[0]);
        let l = 0;
        let h = nums.length - 1;
        while (l < h) {
            if (sorted[l][0] + sorted[h][0] == target) {
                const a = sorted[l][1];
                const b = sorted[h][1];
                return [Math.min(a, b), Math.max(a, b)];
            } else if (sorted[l][0] + sorted[h][0] < target) l ++;
            else h --;
        }

        return [-1, -1]
    }
}
